consensus/types/src/signed_beacon_block.rs

```diff
@@ -510,6 +516,99 @@ impl<E: EthSpec, Payload: AbstractExecPayload<E>> ForkVersionDeserialize
     }
 }
 
+/// This module can be used to encode and decode a `SignedBeaconBlock` the same way it
+/// would be done if we had tagged the superstruct enum with
+/// `#[ssz(enum_behaviour = "union")]`
+/// This should _only_ be used *some* cases when storing these objects in the database
+/// and _NEVER_ for encoding / decoding blocks sent over the network!
+pub mod ssz_tagged_signed_beacon_block {
```

Comment:
I believe so, yes.

Whenever we `decode` a `BeaconBlock` or `BeaconState` with these `from_ssz_bytes` methods which take a `spec`, we do this *only* with those objects as the sole, top-level object. The problem is that the code here requires us to decode these objects *recursively* (when they exist as members of members of other ssz-encoded objects that we are decoding as a whole). The only way to accomplish this (as far as I know) would be to define another trait that mimics the behavior of [`ForkVersionDeserialize`](https://github.com/sigp/lighthouse/blob/stable/consensus/types/src/fork_versioned_response.rs#L48), only for `ssz` instead of `serde`. I'm not even sure if this is possible, and it would be a ton of work to define it for everything even if it were possible.

Reply:
Right I see, ok then I guess this is a relatively simple way to address the problem